server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileComparison.java

```diff
@@ -169,4 +165,21 @@ public String rightSeverity() {
     }
   }
 
+  private enum ActiveRuleToRuleKey implements Function<ActiveRuleDto, RuleKey> {
+    INSTANCE;
+
+    @Override
+    public RuleKey apply(@Nonnull ActiveRuleDto input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "input" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)